### ISSUE TITLE
[Proposal] Flatpak Support

### ISSUE THREAD

**Issue by chenasraf** [Redot-Engine/redot-proposals#22]
There have been several user requests for a Flatpak version, so we should support that additional option for installation.

For context, Flatpak is an app distribution platform for Linux. It has many pros and cons, many users, and still growing in popularity.

I have taken the liberty of creating a Flatpak file for users to be able to install from.

- [The repository](https://github.com/chenasraf/org.redotengine.Redot)
- [Example working job with artifacts](https://github.com/chenasraf/org.redotengine.Redot/actions/runs/11356349863)
- [Godot's original repo on Flathub](https://github.com/flathub/org.godotengine.Godot)

There are a few main things left to discuss:

- The repository ownership - I believe it should be transferred to the org. That will enable easier collaboration and more people can contribute without relying on one person to handle any incoming issues or PRs.
- Submission to Flathub - Here is their [submissions page](https://docs.flathub.org/docs/for-app-authors/requirements), with information about requirements. Most of these are irrelevant but there are still some left to update. We should aim towards release on their Hub for increased availability & discoverability.

## Submission

I've summarized what's left to be changed in order to apply for Flatpak submission.

* [ ] **Stable release** - Flathub only hosts stable releases and not development snapshots or nightly releases - so we need to wait for at least 1 stable release before submission.
* [ ] **flathub.json**: a flathub.json file must be included in the submission if the app only supports one arch of distribution - I'm not sure that's the case for the Redot project (I don't see this file on Godot's original repo) but we should make sure our build process puts this file if necessary.
* [ ] **Screenshots** We need to provide links to the correct blobs (github static links are fine) of screenshots for after the stable release
* [ ] **App data** - we need to update any relevant links in [this file](https://github.com/chenasraf/org.redotengine.Redot/blob/master/org.redotengine.Redot.appdata.xml) and make sure they lead to proper locations
* [ ] **Developer info** - We need to update [this file on Redot's engine repo](https://github.com/Redot-Engine/redot-engine/blob/4.3/misc/dist/linux/org.godotengine.Godot.appdata.xml) and other files in the same directory to properly reflect the rebranding and also include a developer's contact info - can be an org email or a representative that's likely to remain active for the longevity of the project for years to come. I suggest the former

I am opening this issue so we can track and discuss how to proceed.

**Comment by TheShadowOfHassen** [comment]
If you want some help... I know a tiny bit about flatpak, not a lot, but I'm maintaining the Musicpod music player. I'm like a level 2 in flatpak, it's not a lot (most are like level 20,) but I can try to pitch in at least with testing.

**Comment by chenasraf** [comment]
Thanks for reminding me to update

We have a Flatpak package working, but still not on the Hub
We also implemented Snap installation as well, which is on the Snap Store already

You can find more info [here](https://github.com/redot-engine/org.redotengine.Redot)

**Comment by chenasraf** [comment]
Some more information regarding the submissions to Flathub.

- Flathub will need to create a repository under their org, with write access to Redot (related) Contributors 
  - We will probably only have access from the user creating the submission, and will need to discover how to add more users
  - This will require us to consider the relationship between the [org.redotengine.Redot](https://github.com/Redot-Engine/org.redotengine.Redot/) repo and the Flathub repo
  - We might just remove any Flatpak code from there and use it only for Snap, or find a way to use it as the source for the eventual repository they will provide us 
  - We won't be able to simply create a PR for updating this way, most likely we will need to create a workflow or script which updates the code if we decide to go that route, because since it's not a direct fork of our repo but of flathub/flathub, a PR can't be created between 2 unrelated repos
- They need to verify domain ownership. See [Verification](https://docs.flathub.org/docs/for-app-authors/verification/)

   TL;DR - Make a file accessible from  https://redotengine.org/.well-known/org.flathub.VerifiedApps.txt which will contain a token they will provide

**Comment by Shadowblitz16** [comment]
I would like this too please.
